lib/rubocop/cop/rspec/be_eq.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for expectations where `be(...)` can replace `eq(...)`.
+      #
+      # The `be` matcher compares by identity while the `eq` matcher compares
+      # using `==`. Booleans and nil can be compared by identity and therefore
+      # the `be` matcher is preferable as it is a more strict test.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo).to eq(true)
+      #   expect(foo).to eq(false)
+      #   expect(foo).to eq(nil)
+      #
+      #   # good
+      #   expect(foo).to be(true)
+      #   expect(foo).to be(false)
+      #   expect(foo).to be(nil)
+      #
+      class BeEq < Base
+        extend AutoCorrector
+
+        MSG = 'Prefer `be` over `eq`.'
+        RESTRICT_ON_SEND = %i[to not_to to_not].freeze
+
+        # @!method eq_type_with_identity(node)
+        def_node_matcher :eq_type_with_identity, <<-PATTERN
+          (send _ #Runners.all $(send nil? :eq {true false nil_type?}))
```

Comment:
Is there a specific reason why you don't include symbols and integers?

Reply:
The comment https://github.com/rubocop/rubocop-rspec/issues/16#issuecomment-237363015 mentions value objects where it’s valid to override `#==` and have an expectation as e.g. `expect(Score.new(1)).to eq(1)`.